geomstats/_backend/numpy/__init__.py

```diff
@@ -146,19 +164,20 @@ def assignment(x, values, indices, axis=0):
     If a list is given, it must have the same length as indices.
     """
     x_new = copy(x)
-    if not isinstance(indices, list):
-        indices = [indices]
-    if not isinstance(values, list):
-        values = [values] * len(indices)
-    for nb_index, index in enumerate(indices):
-        if not isinstance(index, tuple):
-            index = (index,)
-        if len(index) < len(shape(x)):
-            for n_axis in range(shape(x)[axis]):
-                extended_index = index[:axis] + (n_axis,) + index[axis:]
-                x_new[extended_index] = values[nb_index]
-        else:
-            x_new[index] = values[nb_index]
+
+    use_vectorization = hasattr(indices, '__len__') and len(indices) < ndim(x)
+    if _is_boolean(x):
```

Comment:
Should it be the indices that are tested to be boolean or not?

Reply:
Good catch